ntoskrnl/ke/i386/kiinit.c

```diff
@@ -810,36 +811,27 @@ KiSystemStartup(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
     RtlCopyMemory(&Idt[8], &DoubleFaultEntry, sizeof(KIDTENTRY));
 
 AppCpuInit:
-    //TODO: We don't setup IPIs yet so freeze other processors here.
-    if (Cpu)
+    /* Acquire lock */
+    while (InterlockedBitTestAndSet((PLONG)&KiFreezeExecutionLock, 0))
     {
-        KeMemoryBarrier();
-        LoaderBlock->Prcb = 0;
-
-        for (;;)
-        {
-            YieldProcessor();
-        }
+        /* Loop until lock is free */
+        while ((*(volatile KSPIN_LOCK*)&KiFreezeExecutionLock) & 1);
     }
 
-    /* Loop until we can release the freeze lock */
-    do
-    {
-        /* Loop until execution can continue */
-        while (*(volatile PKSPIN_LOCK*)&KiFreezeExecutionLock == (PVOID)1);
-    } while(InterlockedBitTestAndSet((PLONG)&KiFreezeExecutionLock, 0));
-
     /* Setup CPU-related fields */
     __writefsdword(KPCR_NUMBER, Cpu);
     __writefsdword(KPCR_SET_MEMBER, 1 << Cpu);
     __writefsdword(KPCR_SET_MEMBER_COPY, 1 << Cpu);
     __writefsdword(KPCR_PRCB_SET_MEMBER, 1 << Cpu);
 
-    KiVerifyCpuFeatures(Pcr->Prcb);
+    KiVerifyCpuFeatures(KeGetCurrentPrcb());
```

Comment:
Why is this needed?

Reply:
Because throwing a uninitialized variable as a parameter caused RTC to freak out, i could instead change the Pcr variable here specifically on the other processors. But this was just what i did. I dont mind either way.

Reply:
Ahh, I see, there is the goto that skips it.